sdk/servicebus/service-bus/src/util/utils.ts

```diff
@@ -485,3 +487,96 @@ export type EntityStatus =
   | "Renaming"
   | "Restoring"
   | "Unknown";
+
+/**
+ * An executor for a Promise (or function that returns a Promise) that obeys both a timeout and an
+ * optional AbortSignal.
+ * @param timeoutMs - The number of milliseconds to allow before throwing an AMQP error with condition ServiceUnavailableError
+ * @param timeoutMessage - The message to place in the .description field for the thrown exception for Timeout.
+ * @param abortSignal - The abortSignal associated with containing operation.
+ * @param abortErrorMsg - The abort error message associated with containing operation.
+ * @param value - The value to be resolved with after a timeout of t milliseconds.
+ * @returns {Promise<T>} - Resolved promise
+ *
+ * @internal
+ * @ignore
+ */
+export function waitForTimeoutOrAbortOrResolve<T>(args: {
+  actionFn: () => Promise<T>;
+  timeoutMs: number;
+  timeoutMessage: string;
```

Comment:
(Copied from https://github.com/Azure/azure-sdk-for-js/pull/8994#discussion_r430887218)

<img width="550" alt="Screen Shot 2020-05-28 at 12 49 01 PM" src="https://user-images.githubusercontent.com/16890566/83186630-a97f4f00-a0e1-11ea-8505-46cc4d1b345a.png">

Reply:
Making the user preconstruct the error isn't quite right either. 

After talking with @xirzec about this it appears we throw AbortError's for timeouts in core-http. I swapped over to just throwing AbortError for now.

Reply:
We wouldnt want `AbortError` either. These are considered as non retryable and user triggered. Where as the timeout error (atleast in Service Bus) is a retryable one

Reply:
We could use `OperationTimeoutError` which is what we use in quite a few places for timeout in Service Bus, Event Hubs and the underlying libraries core-amqp and rhea-promise

Reply:
Done